src/dom.js

```diff
@@ -1215,7 +1215,28 @@ export const getCurrentScriptUID : GetCurrentScriptUID = memoize(() => {
         return uid;
     }
 
-    uid = uniqueID();
+    if (script.src) {
+
+        let stringToHash = script.src;
+
+        Object.keys(ATTRIBUTES).forEach(key => {
```

Comment:
@leogedler @bluepnume what do you think about ditching the data attribute allowed list and having the hash be `script.src` + `script.dataset` for all data attributes except uid? Ex:

Here's a script with a couple data attributes. Note how `dataset` converts the `data-` keys into camelCase and drops the prefix.
<img width="1629" alt="Screen Shot 2021-05-11 at 9 04 29 AM" src="https://user-images.githubusercontent.com/534034/117829030-fb0fe380-b237-11eb-9745-cb20709ce128.png">

```js
let stringToHash = script.src;

Object.keys(script.dataset).forEach(key => {
    // ignore uid attribute
    if (key === "uid") {
        return;
    }

    // every other data attribute is added to the stringHash
    stringToHash += script.dataset[key];
});
```